### PR TITLE
cross-platform commands

### PR DESCRIPTION
**Pull request by dirtycajunrice**
### issues
* npm uses cmd.exe on windows which does not support `rm` and `cp`.
* bun is not stable on windows. 
 
### fixes
* `rm -rf` is replaced by `git clean -xdf` (suggestion by winston)
* `cp` - wip
* `bun` is replaced by `pnpm ts-node`

---

## PR-Codex overview
This PR updates package scripts and dependencies for multiple apps. It refactors scripts, updates `tsx` version, and aligns dependencies.

### Detailed summary
- Refactored scripts in `dashboard`, `portal`, and `playground-web` apps.
- Updated `tsx` version to `4.19.0`.
- Aligned dependencies and scripts across apps.

> The following files were skipped due to too many changes: `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/dashboard/package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "private": true,
   "scripts": {
     "preinstall": "npx only-allow pnpm",
-    "dev:no-cache": "rm -rf .next/cache && pnpm run dev",
+    "dev:no-cache": "git clean -xdf && pnpm run dev",
```
Comment (jnsdls) [inline apps/dashboard/package.json:7]: we shouldn't even really need this anymore this was more pre-monorepo

Comment (jnsdls) [reply apps/dashboard/package.json:7]: probably safe to just delete it instead

File: apps/playground-web/package.json [hunk 2]
```diff
@@ -3,12 +3,15 @@
   "version": "0.1.1",
   "private": true,
   "scripts": {
-    "dev": "rm -rf .next && next dev",
+    "predev": "git clean -xdf",
```
Comment (jnsdls) [inline apps/playground-web/package.json:6]: I don't think we should need this

File: apps/playground-web/package.json [hunk 3]
```diff
@@ -3,12 +3,15 @@
...
+    "dev": " next dev",
     "build": "next build",
     "start": "next start",
     "format": "biome format ./src --write",
-    "lint": "biome check ./src && eslint ./src",
-    "fix": "biome check ./src --fix && eslint ./src --fix"
+    "prelint": "biome check ./src",
+    "lint": "eslint ./src",
```
Comment (jnsdls) [inline apps/playground-web/package.json:12]: is the `&&` a problem in win, would prefer not to do the duplication here

File: apps/portal/package.json [hunk 4]
```diff
@@ -65,6 +70,7 @@
     "next-sitemap": "^4.2.3",
     "postcss": "^8",
     "tailwindcss": "^3.4.10",
-    "typescript": "5.5.4"
+    "typescript": "5.5.4",
+    "ts-node": "^10.9.2"
```
Comment (jnsdls) [inline apps/portal/package.json:74]: prefer `tsx` over `ts-node`

File: apps/portal/scripts/createEmptySearchIndex.ts [hunk 5]
```diff
@@ -1,12 +1,9 @@
-import { existsSync, writeFileSync } from "node:fs";
+import { existsSync, writeFileSync } from "fs";
```
Comment (jnsdls) [inline apps/portal/scripts/createEmptySearchIndex.ts:1]: why is this necessary? the explicit `node:<x>` should generally be prefered

**Review by jnsdls — APPROVED** [review]
got some merge conflicts to fix - rest looks good

**Review by gregfromstl — APPROVED** [review]

**Comment by jnsdls** [comment]
* **#4385** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4385?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dirtycajunrice and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Pull request merged**